src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,125 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.Deadline;
+import duke.task.Todo;
+import duke.parser.Parser;
+
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A Storage class loads and writes data into an output file.
+ */
+public class Storage {
+    private String filePath;
+    private SimpleDateFormat formatter;
+
+    /**
+     * Constructs a new Storage object to read and write to a text file stored
+     * on the hard disk.
+     *
+     * @param filePath the file path of the file stored on the hard disk.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+    }
+
+    /**
+     * Returns the file path of the file stored on the hard disk.
+     *
+     * @return the file path.
+     */
+    public String getFilePath() {
+        return filePath;
+    }
+
+    /**
+     * Loads the file from the hard disk and reads it's text. The text would then
+     * be converted into Task objects and stored in an ArrayList for duke to process.
+     *
+     * @return an ArrayList that contains the tasks stored on the hard disk.
+     * @throws DukeException if the file stated in the file path does not exist.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File file = new File(filePath);
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String[] line = sc.nextLine().split(" \\| ");
+                switch (line[0]) {
+                case "T":
+                    Todo todo = new Todo(line[2]);
+                    if (line[1].equals("1")) {
+                        todo.markDone();
+                        tasks.add(todo);
+                    } else {
+                        tasks.add(todo);
+                    }
+                    break;
+                case "D":
+                    Deadline deadline = new Deadline(line[2], Parser.parseDate(formatter, line[3]));
+                    if (line[1].equals("1")) {
+                        deadline.markDone();
+                        tasks.add(deadline);
+                    } else {
+                        tasks.add(deadline);
+                    }
+                    break;
+                case "E":
+                    Event event = new Event(line[2], Parser.parseDate(formatter, line[3]));
+                    if (line[1].equals("1")) {
+                        event.markDone();
+                        tasks.add(event);
+                    } else {
+                        tasks.add(event);
+                    }
```

Comment:
Why not consolidate all the tasks.add(events) ie. call the method once after the switch?

Reply:
That's a good suggestion! I could use polymorphism and use a Task reference to refer to the object instances and add it after the switch.